packages/runtime-dom/src/helpers/useCssVars.ts

```diff
@@ -57,7 +57,15 @@ function setVarsOnVNode(
   if (vnode.shapeFlag & ShapeFlags.ELEMENT && vnode.el) {
     const style = vnode.el.style
     for (const key in vars) {
-      style.setProperty(`--${prefix}${key}`, unref(vars[key]))
+      if (typeof vars[key] === 'string') {
```

Comment:
I think it would make more sense to display a warning if its `== null` in dev only and otherwise keep it. You still want to keep numbers for example

Reply:
It's not necessarily whether the value is null, it's whether it's anything that would never result in a valid CSS property value. For example, if an object is passed for some reason then the CSS property value comes out as [object Object] and the fallback parameter in the CSS var isn't used.

Although you have a point about numbers as they would still work. I think I'd rather try for some consistency on wanting strings as the values, but I can't disagree. Easy enough to check against both string and number.

Reply:
sure for a stricter check of strings and numbers as long as it's dev only and its purpose is to log a warn

Reply:
Would you feel adding a bit more to the warning to explain that it expects a string or number is needed? Right now it just says that the value is being skipped but not why.

Reply:
It could say that it must be a number or a string but xxx was received